tools/sgapigateway/tools.go

```diff
@@ -0,0 +1,99 @@
+package sgapigateway
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"net/http"
+	"os"
+
+	"go.einride.tech/sage/sg"
+)
+
+//go:embed Dockerfile
+var dockerfile []byte
+
+func EndpointsConfigID(ctx context.Context, serviceName, gcpProject string) string {
+	sg.Logger(ctx).Printf(
+		"retrieving endpoints configID from %s in %s...",
+		serviceName,
+		gcpProject,
+	)
+
+	return sg.Output(sg.Command(
+		ctx,
+		"gcloud",
+		"endpoints",
+		"configs",
+		"list",
+		"--service",
+		serviceName,
+		"--project",
+		gcpProject,
+		"--limit",
+		"1",
+		"--format",
+		"value(id)",
+	))
+}
+
+func DockerImage(ctx context.Context, serviceName, gcpProject, gcpRegion string) string {
+	if err := os.MkdirAll(sg.FromToolsDir("api-gateway"), 0o755); err != nil {
```

Comment:
Can we avoid doing this in a folder which may look like it belongs in the project just for building a temporary image? Maybe do it in a temporary disposable folder such as `os.TempDir`?

Reply:
Sure, i dont see why not!

Reply:
Went with that and created a dir i `os.TempDir`